node/core/approval-voting/src/lib.rs

```diff
@@ -2038,23 +2144,35 @@ async fn launch_approval(
 				// No dispute necessary, as this indicates that the chain is not behaving
 				// according to expectations.
 				metrics_guard.take().on_approval_unavailable();
-				return;
+				return ApprovalState::failed(
+					validator_index,
+					candidate_hash,
+					candidate_index as _,
+					InvalidCandidate::InvalidOutputs,
+				);
 			}
 		};
 
 		let (val_tx, val_rx) = oneshot::channel();
 
 		let para_id = candidate.descriptor.para_id;
-		let _ = background_tx.send(BackgroundRequest::CandidateValidation(
+
+		sender.send_message(CandidateValidationMessage::ValidateFromExhaustive(
 			available_data.validation_data,
 			validation_code,
 			candidate.descriptor,
 			available_data.pov,
 			val_tx,
-		)).await;
+		).into()).await;
 
 		match val_rx.await {
-			Err(_) => return,
+			Err(_) =>
+				return ApprovalState::failed(
+					validator_index,
+					candidate_hash,
+					candidate_index as _,
+					InvalidCandidate::InvalidOutputs,
```

Comment:
Isn't this a `oneshot::Canceled`?

Not sure exactly what to do here, but `InvalidCandidate` is definitely wrong